python/setup.py

```diff
@@ -272,6 +272,8 @@ def get_packages(self):
             "fastapi",
             "aiorwlock",
             "watchfiles",
+            "numpy >= 1.16; python_version < '3.9'",
+            "numpy >= 1.19.3; python_version >= '3.9'",
```

Comment:
FYI @ray-project/ray-serve we should try to also drop this as a requirement for Serve once the "HTTP adapters" are fully deprecated